rust-runtime/aws-smithy-http-auth/src/definition.rs

```diff
@@ -142,17 +155,27 @@ pub mod http_auth_definition {
         }
 
         /// Constructs a [`HttpAuthDefinition`] from the builder.
-        pub fn build(self) -> HttpAuthDefinition {
-            if self.scheme.is_some() && !self.name.eq_ignore_ascii_case("authorization") {
+        pub fn build(self) -> Result<HttpAuthDefinition, AuthError> {
+            if self.scheme.is_some()
+                && self
+                    .name
+                    .as_deref()
+                    .map_or("".to_string(), |s| s.to_ascii_lowercase())
+                    != "authorization"
+            {
                 // Stop execution because the Smithy model should not contain such combination.
                 // Otherwise, this would cause unexpected behavior in the SDK.
                 panic!("{}", AuthError::from(AuthErrorKind::SchemeNotAllowed));
             }
-            HttpAuthDefinition {
-                location: self.location,
-                name: self.name,
+            Ok(HttpAuthDefinition {
+                location: self.location.ok_or_else(|| {
+                    AuthError::from(AuthErrorKind::MissingRequiredField("location"))
+                })?,
+                name: self
+                    .name
+                    .ok_or_else(|| AuthError::from(AuthErrorKind::MissingRequiredField("name")))?,
```

Comment:
These also seem like they should panic since they are cases where the builder is being used incorrectly.